EpiserverRedirects/Repository/Repository.cs

```diff
@@ -35,5 +35,13 @@ protected static void WriteToModel(RedirectRule redirectRule, RedirectRule redir
             redirectRuleToUpdate.Priority = redirectRule.Priority;
             redirectRuleToUpdate.ContentId = redirectRule.ContentId;
         }
+
+        protected static void EncodeSpacesInUrls(RedirectRule redirectRule)
+        {
+            if (redirectRule == null) return;
+            
+            redirectRule.OldPattern = redirectRule.OldPattern?.Replace(" ", "%20");
```

Comment:
@m-jedynak I'm not sure about origin of the issue, but shouldn't we encode other chars ( like `!` ) as well?

Reply:
I don't like two things
- we encode just space not other chars (like !, ...). I think we should use method like HttpUtility.UrlEncode, Uri.EscapeDataString() instead of doing just spaces manually

- it's a bad practice to change value in existing object. I mean
    At some point "redirectRule.OldPattern" is invalid url
    Later "redirectRule.OldPattern" holds a valid value.
    
   I would rather validate that on setter or when saving in database. 
  We can talk if it is not clear

Reply:
Additionally I'm not sure if this logic should be in DynamicDataStoreRepository : RedirectRuleRepository
What if we add new repository in future?